spi-proposal-no-transaction.md

```diff
@@ -0,0 +1,256 @@
+SPI (Proposal)
```

Comment:
This proposal does not include `SPI_BEGIN_TRANSACTION` and `SPI_END_TRANSACTION`. Instead it adds a `csPinChange` paramter to the TRANSFER, READ and WRITE commands. If SPI transactions are supported for a given platform such as Arduino the appropriate functions would be used in the implementation behind the scenes. The value of the `csPinChange` parameter would be an indication of when `endTransaction` would be called in the firmware. `beginTransaction` would be called when a TRANSFER, READ or WRITE command is received **AND** endTransaction had been previously called for the same `deviceId` **OR** it was the first TRANSFER, READ or WRITE command received for that device. While this masks a lot of complexity from the user, there is a risk of begin/end transaction getting out of sync.